doc/public/static/examples/ol-mapset-layer.js

```diff
@@ -0,0 +1,123 @@
+Bar</name><Style><IconStyle><scale>0</scale></IconStyle><LabelStyle><color>ff000000</color><scale>1.5</scale></LabelStyle></Style><ExtendedData><Data name="maxZoom"><value>20</value></Data><Data name="textAlign"><value>center</value></Data><Data name="textArray"><value>["Foo Fizz","","\n","","​Bar",""]</value></Data><Data name="textBackgroundFillColor"><value>rgba(196,41,55,1)</value></Data><Data name="textFont"><value>18px lato</value></Data><Data name="textPadding"><value>7,10,5,10</value></Data><Data name="textRotation"><value>-0.02099384556098904</value></Data><Data name="zIndex"><value>300</value></Data></ExtendedData><Point><coordinates>7.694620155159875,48.03693098841495,0</coordinates></Point></Placemark></Document></kml>`;
+
+  const loadKmlButton = document.getElementById('load-kml-button');
+  loadKmlButton?.addEventListener('click', () => {
+    if (!kmlTextArea.value) {
+      alert('Please enter a valid KML');
+      return;
+    }
+    mapsetLayer.plans = [
+      {
+        data: kmlTextArea.value,
+      },
+    ];
+    zoomOnFeatures();
+  });
+
+  // Load plan by ID
+  const idInput = document?.getElementById('id-input');
+  const loadPlanButton = document.getElementById('load-plan-button');
+  loadPlanButton?.addEventListener('click', async () => {
+    const id = idInput.value.trim();
+    if (!id) {
+      alert('Please enter a plan ID');
+      return;
+    }
+    await mapsetLayer.fetchPlanById(id);
```

Comment:
I m not sure if this fetchPlanById function make sense. I f you want a plan you can do:

```
const plan = await mapsetLayer.api.getPlanById(id);
mapsetLayer.plans = [plan]
```

I said that because i do not want to duplicate all functionnalities between the api and the layer so we do not have to maintain a lot a layer's utilities functions.